Fix two lgtm.com alerts: potential resource leaks

sjvs opened the pull request:
This PR addresses two alerts reported by lgtm.com (@lgtmhq):

```HttpDiscovery.java```: https://lgtm.com/projects/g/bitcoinj/bitcoinj/snapshot/df1f776398d196d24e48653c8c952b97370b1e19/files/core/src/main/java/org/bitcoinj/net/discovery/HttpDiscovery.java#V100

```BuildCheckpoints.java```: https://lgtm.com/projects/g/bitcoinj/bitcoinj/snapshot/df1f776398d196d24e48653c8c952b97370b1e19/files/tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java#V214

There are a total of 29 alerts on lgtm.com for bitcoinj - most of them look legit: https://lgtm.com/projects/g/bitcoinj/bitcoinj/alerts/

If you enable automatic code reviews through PR integration, you can actually catch things like this before they get merged into master :smiley:!

sjvs commented:
Had a look at the test failure: not sure how that could have been caused by my changes?

schildbach commented on tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java:
```diff
@@ -211,7 +211,16 @@ private static void writeTextualCheckpoints(TreeMap<Integer, StoredBlock> checkp
+        try {
+            manager = new CheckpointManager(params, fis);
+        } catch (IOException ioe){
+            fis.close();
```
Why not put the fis.close() in a finally block?

sjvs replied:
ugh - weirdly I thought try-finally wasn't introduced until after Java 6, but that's of course try-with-resources. Ignore me. Will fix.

schildbach commented on tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java:
```diff
@@ -211,7 +211,16 @@ private static void writeTextualCheckpoints(TreeMap<Integer, StoredBlock> checkp
 
     private static void sanityCheck(File file, int expectedSize) throws IOException {
-        CheckpointManager manager = new CheckpointManager(params, new FileInputStream(file));
+    	FileInputStream fis = new FileInputStream(file);
```
Can you squash this commit with the indenting fix you already added?

schildbach commented on tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java:
```diff
@@ -211,7 +211,16 @@ private static void writeTextualCheckpoints(TreeMap<Integer, StoredBlock> checkp
+
+        try {
+            manager = new CheckpointManager(params, fis);
+        } catch (IOException ioe){
```
We prefer a space before the opening bracket (if it's at EOL).

sjvs replied:
will fix

schildbach commented on core/src/main/java/org/bitcoinj/net/discovery/HttpDiscovery.java:
```diff
@@ -97,9 +97,20 @@ public HttpDiscovery(NetworkParameters params, Details details,  OkHttpClient cl
+                throw ioe;
+            }
+            
+            if (zip != null){
```
Why the null check? Looks as if zip can't be null here.

schildbach commented on core/src/main/java/org/bitcoinj/net/discovery/HttpDiscovery.java:
```diff
@@ -97,9 +97,20 @@ public HttpDiscovery(NetworkParameters params, Details details,  OkHttpClient cl
+            try {
+                proto = PeerSeedProtos.SignedPeerSeeds.parseDelimitedFrom(zip);
+            } catch (IOException ioe){
+                zip.close(); // will close InputStream as well
```
Again zip.close() could be in a finally block?

schildbach commented:
Thanks for the PR! I added some minor comments.

sjvs commented:
@schildbach: I think I've addressed your comments.

I had a quick look at the other lgtm.com alerts (https://lgtm.com/projects/g/bitcoinj/bitcoinj/alerts/): a lot of them look legit and well worth fixing, but it requires someone who's a bit more confident with the bitcoinj than me (i.e., you?).

Let me know if there's anything else you'd like me to do in this PR, and keep up the good work!

schildbach commented:
I just squashed your commits a bit and merged! Thanks for bearing with me and sorry for the delay.
